limitador/src/prometheus_metrics.rs

```diff
@@ -27,14 +28,52 @@ pub struct PrometheusMetrics {
     registry: Registry,
     authorized_calls: IntCounterVec,
     limited_calls: IntCounterVec,
+    use_limit_name_label: bool,
 }
 
 impl PrometheusMetrics {
     pub fn new() -> Self {
-        let labels = vec![NAMESPACE_LABEL];
+        Self::new_with_options(false)
+    }
+
+    pub fn new_with_counters_by_limit_name() -> Self {
+        Self::new_with_options(true)
+    }
+
+    pub fn incr_authorized_calls(&self, namespace: &Namespace) {
+        self.authorized_calls
+            .with_label_values(&[namespace.as_ref()])
+            .inc();
+    }
+
+    pub fn incr_limited_calls<'a, LN>(&self, namespace: &Namespace, limit_name: LN)
+    where
+        LN: Into<Option<&'a str>>,
```

Comment:
no `AsRef<str>` in the end?

Reply:
In the end I decided not to do this because it makes the code more complicated (problems when inferring types, etc.).